l1-contracts/contracts/bridge/L1ERC20Bridge.sol

```diff
@@ -148,103 +98,76 @@ contract L1ERC20Bridge is IL1Bridge, IL1BridgeLegacy, ReentrancyGuard {
         uint256 _l2TxGasLimit,
         uint256 _l2TxGasPerPubdataByte
     ) external payable returns (bytes32 l2TxHash) {
-        l2TxHash = deposit(_l2Receiver, _l1Token, _amount, _l2TxGasLimit, _l2TxGasPerPubdataByte, address(0));
+        // we need to deposit and set allowance, as the sender set the allowance to this contract
+        uint256 amount = _depositFundsToSharedBridge(msg.sender, _l1Token, _amount);
```

Comment:
It is also seem possible to `claimFailedDeposit` a failed WETH deposit, meaning that user will both have funds on L2 and on L1

Reply:
It should be possible to call `claimFailedDeposit`, but the process should fail, as we should not save 'depositHappened' for baseToken deposits. This is indeed missing